plugins/file-upload/src/index.jsx

```diff
@@ -1,189 +1,166 @@
-
-    const {
-        dialogStyles,
-        headerStyles,
-        contentStyles,
-        footerStyles,
-        cancelButtonStyles
-    } = getStylesMemo(theme);
-
-    return (
-        <div
-            {...attributes}
-            style={{
-                ...attributes.styles,
-                ...dialogStyles
-            }}
-        >
-            {(service === 'amazon-s3' || service === 'azure' || service === 'live-agent') &&
-                (!isUploading && !pluginBusy ? (
-                    <>
-                        <header style={headerStyles}>{titleText || 'File Upload'}</header>
-                        <main style={contentStyles} {...getRootProps()}>
-                            <input {...getInputProps()} />
-                            {isDragActive ? (
-                                <p>{dragClickLabel || 'Drop the file here'}</p>
-                            ) : (
-                                    <p>{dropLabel || 'Drag a file here, or click to select one'}</p>
-                                )}
-                            {isRejected ? (
-                                <SizeLimitError />
-                            ) : null}
-                        </main>
-                        <footer style={footerStyles}>
-                            <button
-                                type='button'
-                                onClick={onDismissFullscreen}
-                                style={cancelButtonStyles}
-                            >
-                                {cancelButtonLabel || 'cancel'}
-                            </button>
-                        </footer>
-                    </>
-                ) : (
-                        <>
-                            <header style={headerStyles}>{titleText || 'File Upload'}</header>
-                            <Spinner theme={theme} />
-                        </>
-                    ))}
-        </div>
-    );
+	const [isUploading, setIsUploading] = React.useState(false);
+	const [isRejected, setIsRejected] = React.useState(false);
+
+	const { isFullscreen, onSetFullscreen, theme, message } = props;
+
+	const { openButtonLabel, sizeLimit, fileSizeLimitErrorMessage } = message.data._plugin;
+	// handles change events for the file input
+	// starts the upload
+	const onDrop = React.useCallback(async files => {
+		const file = files[0];
+
+		if (file.size > sizeLimit * 1000 * 1000) {
+			setIsRejected(true);
+			return;
+		}
+
+		setIsRejected(false);
+		try {
+			setIsUploading(true);
+			pluginBusy = true;
+			const result = await upload(props.message.data._plugin, file);
+			setIsUploading(false);
+			pluginBusy = false;
+			if (result.success) {
+				props.onSendMessage("", {
+					file: result.url,
+				});
+				props.onSendMessage("File upload succeeded", {});
+			} else {
+				props.onSendMessage(
+					`Your file upload failed. Please verify your file is less than 40MB large and is of type jpg, jpeg, png, pdf, doc or docx.`,
```

Comment:
This line will cause any failed file upload from any provider to show this particular error message. This is very generic and not applicable at all. We should rather let the "upload providers" analyze the response / result and generate an error message that gets returned instead of a single static one at the top level

Reply:
fixed

Reply:
fixed